enterprise/frontend/src/embedding-sdk/store/auth/auth.ts

```diff
@@ -16,33 +15,40 @@ import { refreshSiteSettings } from "metabase/redux/settings";
 import { refreshCurrentUser } from "metabase/redux/user";
 import type { Settings } from "metabase-types/api";
 
-import { getOrRefreshSession } from "./reducer";
-import { getFetchRefreshTokenFn } from "./selectors";
+import { getOrRefreshSession } from "../reducer";
+import { getFetchRefreshTokenFn } from "../selectors";
+
+import { jwtRefreshTokenFn } from "./jwt";
+import { popupRefreshTokenFn } from "./popup";
 
 export const initAuth = createAsyncThunk(
   "sdk/token/INIT_AUTH",
   async (authConfig: MetabaseAuthConfig, { dispatch }) => {
     // Setup JWT or API key
-    const isValidAuthProviderUri =
-      authConfig.authProviderUri && authConfig.authProviderUri?.length > 0;
+    const isValidInstanceUrl =
+      authConfig.metabaseInstanceUrl &&
+      authConfig.metabaseInstanceUrl?.length > 0;
     const isValidApiKeyConfig = authConfig.apiKey && getIsLocalhost();
 
-    if (isValidAuthProviderUri) {
-      // JWT setup
+    if (isValidApiKeyConfig) {
+      // API key setup
+      api.apiKey = authConfig.apiKey;
+    } else if (isValidInstanceUrl) {
+      // SSO setup
       api.onBeforeRequest = async () => {
         const session = await dispatch(
-          getOrRefreshSession(authConfig.authProviderUri!),
+          getOrRefreshSession(authConfig.metabaseInstanceUrl),
         ).unwrap();
```

Comment:
If there is an error in the authentication, does it propagate to the SdkError? I assume yes